server/src/main/java/org/apache/druid/server/SegmentManager.java

```diff
@@ -258,6 +316,11 @@ public void dropSegment(final DataSegment segment)
 
               log.info("Attempting to close segment %s", segment.getId());
               oldQueryable.close();
+
+              final ReferenceCountingIndexedTable oldTable = dataSourceState.tablesLookup.remove(segment.getId());
+              if (oldTable != null) {
+                oldTable.close();
```

Comment:
Hmm, should we use `Closer` to close `oldTable` and `oldQueryable` so that both can be closed even when any potential errors is thrown?

Reply:
changed to use `Closer`